src/main/java/net/minecraftforge/client/ItemDecorator.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client;
+
+import net.minecraft.client.gui.Font;
+import net.minecraft.client.renderer.entity.ItemRenderer;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.ItemStack;
+
+import javax.annotation.Nullable;
+
+/**
+ * An Object that is used by the {@Link: net.minecraftforge.items.ItemDecoratorHandler} capability.
+ */
+public abstract class ItemDecorator{
+    private final ResourceLocation key;
```

Comment:
If they key is only used as an identifier in the map, why does it need to be attached to the decorator itself? Couldn't I just pass in the resource key for the map without having to deal with knowing what the decorator is? Otherwise I would always need to keep an instance of the decorator if I ever wanted to remove it based on the current implementation.